### PR TITLE
IconColumn record in options

### PR DESCRIPTION
**Pull request by dejury**
This PR makes it possible to process the current record in the callback of the IconColumn::options(). This makes it much easier to do dynamic icon selection.

```php
use Filament\Tables\Columns\IconColumn;

IconColumn::make('is_featured')
    ->options([
        'heroicon-o-x-circle',
        'heroicon-o-pencil' => fn ($state, $record): bool => $record->status === 2,
        'heroicon-o-clock' => fn ($state): bool => $state === 'reviewing',
        'heroicon-o-check-circle' => fn ($state): bool => $state === 'published',
    ])
```

### CODE REVIEW

**Comment by danharrin** [comment]
Please save the record in a variable above the loop so we dont call `getRecord()` multiple times

**Comment by dejury** [comment]
> Please save the record in a variable above the loop so we dont call `getRecord()` multiple times

Check! Done that

File: packages/tables/docs/03-columns/03-icon.md [hunk 1]
```diff
@@ -16,15 +16,15 @@ IconColumn::make('is_featured')
     ])
 ```
 
-You may also pass a callback to activate an option, accepting the cell's `$state`:
+You may also pass a callback to activate an option, accepting the cell's `$state` & `$record`:
```
Comment (danharrin) [inline packages/tables/docs/03-columns/03-icon.md]:
```suggestion
You may also pass a callback to activate an option, accepting the cell's `$state` and `$record`:
```

**Comment by danharrin** [comment]
Thanks

**Pull request merged**